pkg/controller/compute/gke_test.go

```diff
@@ -109,6 +110,7 @@ func TestSyncClusterGetError(t *testing.T) {
 	r := &Reconciler{
 		Client:   NewFakeClient(tc),
 		resolver: &mockReferenceResolver{},
+		log:      logging.NewNopLogger(),
```

Comment:
Hmm I feel like there could be a better way of doing this, but I don't have any ideas yet.

Reply:
Could you elaborate on why you think this pattern is suboptimal? I believe it's fairly common in Go to use a no-op instrumentation implementation that can be replaced with a functional one via arguments. It's a pattern I've personally used in almost every Go project I've worked on.

Reply:
Oh - or are you specifically calling out the plumbing we're having to do in these unit tests? If so I agree that there's a better way to do this - see what I've done with `managed.Reconciler` and friends in the corresponding `crossplane-runtime` PR. I was not inclined to refactor the few legacy managed resource controllers that still exist to use this better pattern - I'd prefer to replace them with `managed.Reconciler` in future PRs.